loki/transformations/single_column/hoist.py

```diff
@@ -56,6 +56,7 @@ def driver_variable_declaration(self, routine, variables):
             )
 
         block_var = get_integer_variable(routine, self.block_dim.size)
+        block_var = remap_variables(routine, variables=block_var)[0]
```

Comment:
Would it be easy to capture the effect of this in one of the tests for hoist? This seems important enough to warrant capturing this pattern in the test base.